docs-site/content/0.19.0/guide/features/faceting.md

```diff
@@ -1 +1,215 @@
+
+<Tabs :tabs="['JavaScript','PHP','Python','Ruby']">
+  <template v-slot:JavaScript>
+
+```javascript
+let searchParameters = {
+  'q'         : 'experiment',
+  'query_by'  : 'title',
+  'facet_by' : 'authors_facet',
+  'sort_by'   : 'average_rating:desc'
+}
+
+client.collections('books')
+  .documents()
+  .search(searchParameters)
+  .then(function (searchResults) {
+    console.log(searchResults)
+  })
+```
+  </template>
+
+  <template v-slot:PHP>
+
+```php
+$searchParameters = [
+  'q'         => 'experiment',
+  'query_by'  => 'title',
+  'facet_by'  => 'authors_facet',
+  'sort_by'   => 'average_rating:desc'
+];
+
+$client->collections['books']->documents->search($searchParameters);
+```
+  </template> 
+  <template v-slot:Python>
+
+```python
+search_parameters = {
+  'q'         : 'experiment',
+  'query_by'  : 'title',
+  'facet_by' : 'authors_facet',
+  'sort_by'   : 'average_rating:desc'
+}
+
+client.collections['books'].documents.search(search_parameters)
+```
+   </template>
+   <template v-slot:Ruby>
+
+```ruby
+search_parameters = {
+  'q'         => 'experiment',
+  'query_by'  => 'title',
+  'facet_by'  => 'authors_facet',
+  'sort_by'   => 'average_rating:desc'
+}
+
+client.collections['books'].documents.search(search_parameters)
+```
+  </template>
+</Tabs>
+
+The `facet_by` parameter provides the results grouped by authors. Here is a sample response:
+
+```json
+{
+  "facet_counts": [
+    {
+      "field_name": "authors_facet",
+      "counts": [
+          {
+              "count": 2,
+              "value": " Käthe Mazur"
+          },
+          {
+              "count": 2,
+              "value": "Gretchen Rubin"
+          },
+          {
+              "count": 2,
+              "value": "James Patterson"
+          },
+          {
+              "count": 2,
+              "value": "Mahatma Gandhi"
+          }
+      ]
+    }
+  ],
+  "found": 3,
+  "hits": [
+    {
+      "_highlight": {
+        "title": "The Angel <mark>Experiment</mark>"
+      },
+      "document": {
+        "authors": [
+            "James Patterson"
+        ],
+        "authors_facet": [
+            "James Patterson"
+        ],
+        "average_rating": 4.08,
+        "id": "569",
+        "image_url": "https://images.gr-assets.com/books/1339277875m/13152.jpg",
+        "publication_year": 2005,
+        "publication_year_facet": "2005",
+        "ratings_count": 172302,
+        "title": "The Angel Experiment"
+      }
+    },
+    ...
+  ]
+}
+```
+
+As you see in the response, by using faceted results TypeSense can group the books by authors and then find the most relevant result. It also returns the number of books written by each author in the `facet_counts` fields. You can limit the number of faceted results using `max_facet_value` parameter. More details on these arguments can be found [here](https://typesense.org/docs/0.19.0/api/documents.html#arguments).
```

Comment:
> by using faceted results TypeSense can group the books by authors and then find the most relevant result.

by using `facet_by` Typesense can group the results by authors and show you counts of how many of the books in the result set were written by each author.